Document the release flag support (#13668)
gradle/gradle: subprojects/docs/src/docs/userguide/dep-man/04-modeling-features/cross_project_publications.adoc

When you apply the `java` or `java-library` plugins, Gradle will automatically associate this attribute to the outgoing variants.
This means that all libraries published with Gradle automatically tell which target platform they use.

> By default, the `org.gradle.jvm.version` is set to the value of the <<building_java_projects#sec:java_cross_compilation,`release` property>>, if set, or the _target compatibility_ of the main compilation task fo the source set.

Reviewer: ```suggestion
By default, the `org.gradle.jvm.version` is set to the value of the <<building_java_projects#sec:java_cross_compilation,`release` property>> (or as fallback to the `targetCompatibility` value) of the main compilation task of the source set.
```

(Added "fallback" to clarify when the "or" case is happening)
